packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -561,7 +561,8 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
       style: _textStyle,
     );
 
-    Navigator.push(context, _dropdownRoute).then<Null>((_DropdownRouteResult<T> newValue) {
+    Navigator.push(context, _dropdownRoute).then<Null>((Object poppedValue) {
+      final _DropdownRouteResult<T> newValue = poppedValue;
```

Comment:
Can we make `push` generic enough to avoid this change?

Reply:
I thought about trying this.  From a quick glance, I think you can make it more typed, but I'm pretty sure there's some place where you bottom out on untyped, since the presumably the stack has different types on it, and I don't think you can model that in Dart.

Reply:
Done.